libraries/Microsoft.Bot.Builder.Prompts/OAuthPrompt.cs

```diff
@@ -0,0 +1,173 @@
+                throw new InvalidOperationException("OAuthPrompt.Prompt(): length of attachments cannot be null");
+
+            var cards = activity.Attachments.Where(a => a.Content is OAuthCard);
+            if (cards.Count() == 0)
+                throw new InvalidOperationException("OAuthPrompt.Prompt(): atleast one of the cards should be an oauth card");
+
+            var replyActivity = MessageFactory.Attachment(cards.First());//todo:send an oauth or signin card based on channel id
+            await context.SendActivity(replyActivity);
+        }
+
+        public async Task Prompt(ITurnContext context, string text, string speak = null)
+        {
+            await context.SendActivity(text, speak);
+            var adapter = context.Adapter as BotFrameworkAdapter;
+            if (adapter == null)
+                throw new InvalidOperationException("OAuthPrompt.Prompt(): not supported by the current adapter");
+
+            Attachment cardAttachment = null;
+
+            if (asSignInCard(context.Activity.ChannelId))
+            {
+                var link = await adapter.GetOauthSignInLink(context, _settings.ConnectionName); 
+                cardAttachment = new Attachment()
+                {
+                    ContentType = SigninCard.ContentType,
+                    Content = new SigninCard()
+                    {
+                        Text = _settings.Text,
+                        Buttons = new CardAction[]
+                        {
+                        new CardAction() {Title = _settings.Title, Value = link, Type = ActionTypes.Signin }
+                        }
+                    }
+                };
+            }
+            else
+            {
+                cardAttachment = new Attachment()
+                {
+                    ContentType = OAuthCard.ContentType,
+                    Content = new OAuthCard()
+                    {
+                        Text = _settings.Text,
+                        ConnectionName = _settings.ConnectionName,
+                        Buttons = new CardAction[]
+                        {
+                        new CardAction() {Title = _settings.Title, Text = _settings.Text, Type = ActionTypes.Signin }
+                        }
+                    }
+                };
+            }
+            var replyActivity = MessageFactory.Attachment(cardAttachment);
+            await context.SendActivity(replyActivity);
+        }
+
+        public async Task<TokenResult> Recognize(ITurnContext context)
+        {
+            if (IsTokenResponseEvent(context))
+            {
+                var tokenResponse = context.Activity.Value as TokenResponse;
+                return new TokenResult() { Status = PromptStatus.Recognized, Value = tokenResponse };
+
+            }
+            else if (context.Activity.Type == ActivityTypes.Message)
+            {
+                var matched = regex.Match(context.Activity.Text);
```

Comment:
Comments here would be great, as would an example. I have no idea what this regex is trying to match against, and what's valid/invalid.